forte/data/data_store.py

```diff
@@ -187,6 +197,76 @@ def _new_tid(self) -> int:
         r"""This function generates a new ``tid`` for an entry."""
         return uuid.uuid4().int
 
+    def _add_new_type(self, type_name: str):
```

Comment:
Further the name is a bit misleading. It is more like `_get_type_info`, since we never really add a new type, they are either in the dict or in python class.